doc/dev/release_checklist.md

```diff
@@ -0,0 +1,135 @@
+# Releasing a stable version
+
+This checklist walks you through releasing a stable version.
+See below for the checklist for release candidates.
+
+We'll use `v4.6.0` as the intended release version as a running example.
+
+- `git checkout releases/v4.6.0`
+  (This branch should already exist, from the release candidates.)
+- `git pull`
+- In `src/CMakeLists.txt`, verify you see
+  - `set(LEAN_VERSION_MINOR 6)` (for whichever `6` is appropriate)
+  - `set(LEAN_VERSION_IS_RELEASE 1)`
+  - (both of these should already be in place from the release candidates)
+- Run `git diff master RELEASES.md`.
+  - You should expect to see changes in the `v4.7.0-rc1` section, which you may ignore.
+    (i.e. the new release notes for the upcoming release candidate).
+  - But if there are discrepancies in the `v4.6.0` section, you should reconcile these, either by:
+    - Using `git cherry-pick` to pull the commits that modified the release notes on `master`
+    - Simply pushing a new commit to `releases/v4.6.0`.
```

Comment:
A one-sentence summary of the purpose of this step would be useful. Is it ensuring that fixes from the RC are backported to the release branch?

Reply:
Done.